src/d3a/models/strategy/storage.py

```diff
@@ -74,13 +77,25 @@ def __init__(self, initial_soc: float = StorageSettings.MIN_ALLOWED_SOC,
                                  fit_to_limit=fit_to_limit,
                                  energy_rate_change_per_update=energy_rate_decrease_per_update,
                                  update_interval=update_interval)
+        for time_slot in generate_market_slot_list():
+            result = validate_storage_device(
```

Comment:
How about renaming the `result` variable to `is_valid` (every time you loop over device parameters, also in PV strategy)?